premake: new, 5.0.0~beta1

xyq1113723547 opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------
premake: new, 5.0 alpha 16
<!-- Please input topic description here. -->

Package(s) Affected
-------------------

<!-- Please list all package(s) affected by this topic here. -->

Security Update?
----------------

<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

<!-- Yes - Issue Number: ISSUENUMBER -->
No

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->

<!--
Build Order
-----------

Please describe in what order maintainers should build this pull request.
-->

Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

<!-- TODO: CI to auto-fill architectural progress. -->

Commit:
premake: new, 5.0~a16

KexyBiscuit commented on extra-utils/premake/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+# build script copied from Arch Linux
+abinfo "Building premake ..."
+cd "build/gmake2.unix"
+make STRIP="ture" config=release
```
`ture`?

KexyBiscuit commented on extra-utils/premake/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+cd "build/gmake2.unix"
+make STRIP="ture" config=release
+
+abinfo "Installing premake and LISENSE"
```
`LISENSE`?

KexyBiscuit commented on extra-utils/premake/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+
+abinfo "Installing premake and LISENSE"
+cd ../..
+install -Dm755 "bin/release/premake5" "${PKGDIR}/usr/bin/premake5"
```
Use verbose. `-v`

MingcongBai replied:
Absolute path.

KexyBiscuit commented on extra-utils/premake/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+# build script copied from Arch Linux
+abinfo "Building premake ..."
+cd "build/gmake2.unix"
```
Absolute path everywhere, like `cd "$SRCDIR"`

MingcongBai replied:
Also, use parenthetical expression.

```bash
(
    cd "$SRCDIR"/build/gmake2.unix
    ...
)

KexyBiscuit commented on extra-utils/premake/autobuild/defines:
```diff
@@ -0,0 +1,5 @@
+PKGNAME=premake
+PKGSEC=utils
+PKGDEP="glibc"
```
No need to list `glibc` here.

MingcongBai commented on extra-utils/premake/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+abinfo "Installing premake and LISENSE"
+cd ../..
+install -Dm755 "bin/release/premake5" "${PKGDIR}/usr/bin/premake5"
+install -Dm644 LICENSE.txt "$PKGDIR/usr/share/licenses/$PKGNAME/LICENSE"
```
Verbosity here, also absolute path, as with everywhere else.

MingcongBai commented on extra-utils/premake/spec:
```diff
@@ -0,0 +1,5 @@
+VER=5.0~a16
```
Use 5.0.0~alpha16, and `${VER/\~/-}` in `SRCS`.

xyq1113723547 committed:
premake: fix absolute path and typo

MingcongBai commented on extra-utils/premake/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+# build script copied from Arch Linux
+abinfo "Building premake ..."
+cd "$PKGDIR/build/gmake2.unix"
```
Make sure you understand what `$PKGDIR` and `$SRCDIR` are respetively.

MingcongBai commented on extra-utils/premake/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+make STRIP="true" config=release
+
+abinfo "Installing premake and LISENSE file"
+cd ../..
```
Please just use parenthetical expression instead, cd-ing back is a waste of keyboard.

xyq1113723547 committed:
premake: fix typo and add verbose flag

MingcongBai commented on extra-utils/premake/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+
+abinfo "Installing premake and LISENSE file"
+cd "$SRCDIR"
+install -v -Dm755 "bin/release/premake5" "${PKGDIR}/usr/bin/premake5"
```
Use absolute path here.

KexyBiscuit replied:
Merge the options.
```suggestion
install -Dvm755 "bin/release/premake5" "${PKGDIR}/usr/bin/premake5"
```

KexyBiscuit replied:
Use absolute path everywhere.
```suggestion
install -v -Dm755 "$SRCDIR/bin/release/premake5" "${PKGDIR}/usr/bin/premake5"
```

MingcongBai commented on extra-utils/premake/autobuild/build:
```diff
@@ -0,0 +1,9 @@
+abinfo "Installing premake and LISENSE file"
+cd "$SRCDIR"
+install -v -Dm755 "bin/release/premake5" "${PKGDIR}/usr/bin/premake5"
+install -v -Dm644 LICENSE.txt "$PKGDIR/usr/share/licenses/$PKGNAME/LICENSE"
```
Use absolute path here.

KexyBiscuit replied:
```suggestion
install -Dvm644 LICENSE.txt "$PKGDIR/usr/share/licenses/$PKGNAME/LICENSE"
```

KexyBiscuit replied:
```suggestion
install -v -Dm644 "$SRCDIR"/LICENSE.txt "$PKGDIR/usr/share/licenses/$PKGNAME/LICENSE"
```

xyq1113723547 committed:
premake: change VER

xyq1113723547 committed:
premake: Apply suggestions from code review

Co-authored-by: Kexy Biscuit <kexybiscuit@aosc.io>

xyq1113723547 committed:
premake: Apply suggestions from code review

Co-authored-by: Kexy Biscuit <kexybiscuit@aosc.io>

xyq1113723547 committed:
premake: add absolute path

xyq1113723547 committed:
premake: fix typo

KexyBiscuit commented:
/build premake

xyq1113723547 committed:
premake: use parenthetical expression

MingcongBai commented on extra-utils/premake/autobuild/build:
```diff
@@ -0,0 +1,12 @@
+
+abinfo "Installing premake ..."
+(
+  cd "${SRCDIR}"
```
This is not necessary, Autobuild3 runs the build scripts in "$SRCDIR".

xyq1113723547 committed:
premake: delete redundant cd

MingcongBai commented on extra-utils/premake/autobuild/build:
```diff
@@ -0,0 +1,11 @@
+)
+
+abinfo "Installing premake ..."
+(
```
Why is this even necessary here?

xyq1113723547 replied:
Isolating build and install set of commands is beneficial.
If there are errors in make stage, this info will not print out, which suggests errors are in pre-install stage.

KexyBiscuit committed:
premake: lint build script

- build/gmake2.unix/Makefile sets config to release if none is set.
- "STRIP" is nowhere to find in the makefiles nor Arch Linux's PKGBUILD.

KexyBiscuit commented:
/build premake

WhiredPlanck commented on extra-utils/premake/autobuild/defines:
```diff
@@ -0,0 +1,5 @@
+PKGSEC=utils
+PKGDEP="glibc"
+PKGDES="A simple build configuration tool for generating Makefiles"
+ABSPLITDBG=0
```
Please split debug info as much as possible, or explain why it's impossible to be done with a comment. See [`ABSPLITDBG` section](https://wiki.aosc.io/developer/packaging/autobuild3-manual/#:~:text=1%20or%20%E2%80%9Con%E2%80%9D\).-,absplitdbg,-expects%20a%20binary) below [Autobuild3 Manual - Clang specific configurations](https://wiki.aosc.io/developer/packaging/autobuild3-manual/#clang-specific-configurations).

WhiredPlanck commented:
LGTM on AMD64

![图片](https://user-images.githubusercontent.com/47623588/140598872-7c3a2db0-08d4-4c8b-b1cb-d3a3ea845409.png)

WhiredPlanck committed:
premake: update to 5.0.0~beta1

WhiredPlanck committed:
premake: enable spliting debug info

WhiredPlanck committed:
premake: (Fedora spec) enable verbose mode

WhiredPlanck commented:
> LGTM on AMD64
> 
> ![图片](https://user-images.githubusercontent.com/47623588/140598872-7c3a2db0-08d4-4c8b-b1cb-d3a3ea845409.png)

This test process and results are still valid, only the version number has changed.
